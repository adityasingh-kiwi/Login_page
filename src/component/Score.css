.score-username{
    color: greenyellow;
    background-color: cyan;
    text-align: center;
}
.score-markstore{
    color:blue;
    background-color: cyan;
    text-align: center;
}
.score-thanku{
    
    color:red;
    background-color:blue;
    text-align: center;

}