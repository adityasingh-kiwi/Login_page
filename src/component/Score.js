import React, { useEffect } from 'react'
import { useNavigate } from 'react-router';
import { useState } from 'react/cjs/react.development'
import './Score.css'

const Score = () => {
   const [markStore,setMarkStore]=useState();
   const [getUserName,setGetUserName]=useState("")

     
      useEffect(()=>{
          

            let getMarks = localStorage.getItem("Score");
            let getQuestion = localStorage.getItem("numberofQuestion");
            let userName = localStorage.getItem("data");
            var num= ((getMarks*100)/(getQuestion))
              var marks =num.toFixed(2);
            setMarkStore(marks);
            setGetUserName(userName);
   

      },[])
       


      let Naviagte= useNavigate();

   const handleHomePage=()=>{
    Naviagte("/home");
   }

  

    return (
        <div>

            <h1 className="score-thanku">THANK YOU</h1>
            <div>
                <h2 className="score-username">Emaild id: {getUserName}</h2>
            </div>
            <div>
                <h2 className="score-markstore">Scored :{markStore}%</h2>
            </div>
            <div>
                <button className="score-back-to-login" onClick={handleHomePage}>Home page</button>
            </div>
          

        </div>
    )
}
export default Score
