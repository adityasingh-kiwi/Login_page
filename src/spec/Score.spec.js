import Score from "../component/Score";

describe('Score_page',()=>{
    it("Homepage_button_test",()=>{
        let wrapper=shallow(<Score/>)
        expect(wrapper.exists('#home')).toEqual(true);
        wrapper.find('#home').simulate('click');

    })
    it('Snapshot_Score',()=>{
        let wrapper=mount(<Score/>)
        expect(wrapper).toMatchSnapshot();
    })
    it("Homepage_button_test",()=>{
        let wrapper=shallow(<Score/>)
        expect(wrapper.exists('#home')).toEqual(true);
         wrapper.find('#home').simulate('click');
         expect(
             wrapper.find('#home').prop('defaultValue'),
             ).toBeTruthy();

    })
})