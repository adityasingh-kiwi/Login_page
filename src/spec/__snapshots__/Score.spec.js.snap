// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score_page Snapshot_Score 1`] = `
<Score>
  <div>
    <h1
      className="score-thanku"
    >
      THANK YOU
    </h1>
    <div>
      <h2
        className="score-username"
      >
        Emaild id: 
      </h2>
    </div>
    <div>
      <h2
        className="score-markstore"
      >
        Scored :
        NaN
        %
      </h2>
    </div>
    <div
      className="homePage"
    >
      <button
        className="score-back-to-login"
        defaultValue={false}
        id="home"
        onClick={[Function]}
      >
        Home page
      </button>
    </div>
  </div>
</Score>
`;
